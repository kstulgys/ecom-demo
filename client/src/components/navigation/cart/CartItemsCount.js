import React, { Component } from 'react'
import { Button, Pane, Text, Heading, Pill, Spinner, Popover } from 'evergreen-ui'
import posed, { PoseGroup, tween } from 'react-pose'
// import User from '../../user/User'

// const Panel = posed.div({
//   fullscreen: {
//     width: '100vw',
//     height: '100vh',
//     transition: tween,
//     flip: true
//   },
//   thumbnail: {
//     width: 100,
//     height: 100,
//     transition: tween,
//     flip: true
//   }
// });

const Count = posed.span({
	enter: {
		y: -15,
		scale: 1.2
		// transition: tween,
		// flip: true
		// transform: rotateX(0.5),
		// transition: { duration: 100,  }

		// delay: 300
		// transition: {
		// 	y: { type: 'spring', stiffness: 1000, damping: 15 },
		// 	default: { duration: 300 }
		// }
	},
	exit: {
		y: 0,
		scale: 1
		// transition: tween,
		// flip: true
		// transition: { duration: 500 }
	}
})

class CartItemsCount extends Component {
	state = {
		pose: 'exit'
	}
	componentWillReceiveProps(prev, next) {
		this.setState({
			pose: 'enter'
		})
		setTimeout(() => {
			this.setState({
				pose: 'exit'
			})
		}, 90)
	}
	render() {
		return (
			<Count pose={this.state.pose} key={this.props.count}>
				<Pill display="inline-flex" marginLeft={8} backgroundColor="#D9822B" isSolid>
					{this.props.count ? this.props.count : 0}
				</Pill>
			</Count>
		)
	}
}

export default CartItemsCount
