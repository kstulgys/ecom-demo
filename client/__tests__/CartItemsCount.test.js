import React from 'react'
import CartItemsCount from '../src/components/navigation/cart/CartItemsCount'
import toJSON from 'enzyme-to-json'
import { shallow, mount, render } from 'enzyme'

describe('<CartItemsCount/>', () => {
  it('render', () => {
    shallow(<CartItemsCount count={10} />)
  })

  it('matches the snapshot', () => {
    const wrapper = shallow(<CartItemsCount count={11} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('updates via props', () => {
    const wrapper = shallow(<CartItemsCount count={50} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
    wrapper.setProps({ count: 10 })
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
