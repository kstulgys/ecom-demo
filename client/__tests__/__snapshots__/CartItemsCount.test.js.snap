// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItemsCount/> matches the snapshot 1`] = `
<Component
  key="11"
  pose="exit"
>
  <Pill
    backgroundColor="#D9822B"
    display="inline-flex"
    isSolid={true}
    marginLeft={8}
  >
    11
  </Pill>
</Component>
`;

exports[`<CartItemsCount/> updates via props 1`] = `
<Component
  key="50"
  pose="exit"
>
  <Pill
    backgroundColor="#D9822B"
    display="inline-flex"
    isSolid={true}
    marginLeft={8}
  >
    50
  </Pill>
</Component>
`;

exports[`<CartItemsCount/> updates via props 2`] = `
<Component
  key="10"
  pose="enter"
>
  <Pill
    backgroundColor="#D9822B"
    display="inline-flex"
    isSolid={true}
    marginLeft={8}
  >
    10
  </Pill>
</Component>
`;
